### ISSUE TITLE
When remote does not have mosh-server, drop a statically-linked mosh-server in /tmp over the SSH connection.

### ISSUE THREAD

**Issue by Gaelan** [mobile-shell/mosh#880]
I'd assume that most servers that people mosh into are running i86 Linux, so it should be possible to have a statically linked mosh-server binary that we copy into /tmp (or ~/bin, so we don't need to do it every time) when we connect to a server without mosh installed. Alternatively, we could just build mosh on the server, which would be slower (and require the remote to have a compiler), but would be compatible with other architectures/OSes, and remove the necessity to distribute a binary blob with mosh.

**Comment by javabrett** [comment]
More or less duplicates #188, #308, #536 .

I also used to long for this feature (using mosh to connect to many servers, which don't usually have mosh-server installed up-front) and it would be great, but per discussions on other issues and on the mailing list, it is fraught with security concerns, before even getting into the complexities of statically-linking.  Discussions such as [SO1](https://stackoverflow.com/questions/2725255/create-statically-linked-binary-that-uses-getaddrinfo) and [SO2](https://stackoverflow.com/questions/5426420/linker-warnings-while-building-application-against-mysql-connector-c-libmysqlcli) highlight the problems with statically-linking where GLIBC and NSS are involved. See also the arguments in [no static linking](https://www.akkadia.org/drepper/no_static_linking.html).

So whilst I think it would be a great feature, I don't expect it to gain acceptance.  I built a statically-linked mosh-server once, and immediately ran into NSS runtime problems as expected.
